### DISCUSSION TITLE
Strange error when pressing backspace (previous page)

### DISCUSSION THREAD

**Discussion by spapas** [unpoly/unpoly#601] [Q&A] [answered]
Hello, i've got some pages that I have enabled up-follow for them; when I click the page links it works fine but when I press bacspace to return to the previous page I get the following error:

```
Uncaught up.Error: up.on(): The "up:click" callback function(t,n){if(!o(t))return;const i=e.attr(n,"up-emit"),s=e.jsonAttr(n,"up-emit-props"),r=u(t,i);Object.assign(r,s),up.emit(n,r)} cannot be registered more than once
```

and unpoly stops working (until I do a full refresh). I searched google but didn't find anything on this error.

Any idea / tip what could be causing this?

**Comment by triskweline** [comment, marked as answer]
My guess is that your `<script>` elements live inside the `<body>`. Going back replaces the `<body>` by default. This causes your scripts to be run a second time.

Use one of the following methods to address this:

- Move your scripts into the `<head>` and [use `<script defer>`](https://makandracards.com/makandra/504104-you-should-probably-load-your-javascript-with-script-defer)
- Set `up.history.config.restoreTargets` to a selector that does not include your scripts, e.g. a `<main>` element.
- Disable execution of scripts within updated fragments, using `up.fragment.config.runScripts = false`

**Comment by spapas** [comment]
Hello @triskweline you are right! The problem was that I had put my scripts at the end of the <body>... It is a little difficult to move them so I used the `up.history.config.restoreTargets = "main"` config and now it works great!
